src/pages/gas/helpers.ts

```diff
@@ -0,0 +1,38 @@
+export function formatToSignificantDecimals(n: number): number {
+  if (n === 0) return 0;
+  const scale = Math.pow(10, 2 - Math.ceil(Math.log10(Math.abs(n))));
+  const result = Math.round(n * scale) / scale;
+
+  return parseFloat(result.toFixed(20).replace(/\.?0+$/, ''));
+}
+
+export function calculatePriceNativeForCurrentFromToken(
+  nativeTokenPriceInSelectedToken: number,
+  nativeTokenPrice: number,
+  selectedFromTokenPriceUSD: number,
+  decimals: number,
+): number {
+  if (nativeTokenPrice && selectedFromTokenPriceUSD && !isNaN(Number(nativeTokenPriceInSelectedToken))) {
+    const nativeTokenPriceUSD = selectedFromTokenPriceUSD * Number(nativeTokenPriceInSelectedToken);
+    const native = nativeTokenPriceUSD / nativeTokenPrice;
+
+    return Number(native.toFixed(decimals));
+  }
+  return 0;
+}
+
+export function calculateForCurrentFromToken(
+  native: number,
+  nativeTokenPrice: number,
+  selectedFromTokenPriceUSD: number,
+  decimals: number,
+): number {
+  if (nativeTokenPrice && selectedFromTokenPriceUSD && !isNaN(Number(native)) && native) {
+    const nativeTokenPriceUSD = nativeTokenPrice * Number(native);
```

Comment:
The buyout fails for the minimum amount because we need to add the `SellFee` to the price here. You can get the fee from chain with `api.consts.treasuryBuyoutExtension.sellFee`. 

Something like
```
const nativeTokenPriceUSD = nativeTokenPrice * Number(native) * (1 + sellFee);
```

should do the trick, assuming that you first convert the `sellFee` from a raw (integer) to the proper decimal value. 

You can find the code for reference [here](https://github.com/pendulum-chain/pendulum/blob/dd8eb3d55ac2933bfc375a13b7a9d2d73ca4ea86/pallets/treasury-buyout-extension/src/lib.rs#L369-L373).